### PR TITLE
Update credential single select to PageFormSingleSelectAwxResource

### PR DESCRIPTION
**Pull request by marshmalien**
https://issues.redhat.com/browse/AAP-24530
https://issues.redhat.com/browse/AAP-25028

Screenshot 2024-06-05 at 2 03 20 PM

### CODE REVIEW

**Review by akus062381 — APPROVED** [review]

File: frontend/awx/resources/sources/InventorySourceSubForm.tsx [hunk 1]
```diff
@@ -42,15 +42,23 @@ export function InventorySourceSubForm() {
...
             label={t('Credential')}
-            placeholder={t('Add credential')}
-            labelHelpTitle={t('Credential')}
             labelHelp={t(
               'Select credentials for accessing the nodes this job will be ran against. You can only select one credential of each type. For machine credentials (SSH), checking "Prompt on launch" without selecting credentials will require you to select a machine credential at run time. If you select credentials and check "Prompt on launch", the selected credential(s) become the defaults that can be updated at run time.'
             )}
-            isRequired={sourceTypes.slice(1).includes(source as string)}
-            credentialIdPath="credentialIdPath"
-            sourceType={source as string}
-            acceptableCredentialKinds={['kubernetes', 'cloud']}
```
Comment (nixocio) [inline frontend/awx/resources/sources/InventorySourceSubForm.tsx:53]: This was not working if it is single select, but I believe do want this to filter. We need to double check.

Comment (marshmalien) [reply frontend/awx/resources/sources/InventorySourceSubForm.tsx:53]: https://github.com/ansible/ansible-ui/pull/2455/commits/89358f75e363c99805ab6fc3f606d6c7a22e1053 Added kubernetes in follow-up commit

**Review by vidyanambiar — APPROVED** [review]

**Review by vidyanambiar — APPROVED** [review]
lgtm!

**Comment by nixocio** [comment]
As a team we decided to merge this one, since the test failures are not related.

**Pull request merged**